Sources/Kitura/RouterRequest.swift

```diff
@@ -102,13 +93,11 @@ public class RouterRequest {
     /// The currently matched section of the URL.
     public internal(set) var matchedPath = ""
 
-    /// The original URL as a string.
-    public var originalURL: String {
-        return serverRequest.urlString
-    }
+    /// The URL from the request if properly received
+    public internal(set) var url : URL?
 
-    /// The URL.
-    public let url: String
+    /// The URL from the request as URLComponents
+    public internal(set) var urlComponents = URLComponents()
```

Comment:
We do want to fix the deprecated variables (url, urlString, parsedURL) so they contain the full URL, not just the path like they do currently, correct? This can potentially break code that is relying on these variables to only contain the path, but really those variables are currently broken as they do not return what they claim.